pkg/capture/capture_manager.go

```diff
@@ -1,436 +1,355 @@
+	logger, t0 := logging.FromContext(ctx), time.Now()
+
+	// To avoid any interference the update() logic is protected as a whole
+	// This also allows us to interace with the captures without copying (creating potential races)
+	cm.Lock()
+	defer cm.Unlock()
+
+	// Enable any interfaces present in the positive list
 	var rg RunGroup
-	if len(ifaces) > 0 {
-		for _, iface := range ifaces {
-			iface := iface
-			mc, exists := cmCopy[iface]
-			if exists {
-				rg.Run(func() {
-					f := mc.capture.Flows()
-					ifaceFlowsMutex.Lock()
-					ifaceFlows[iface] = f
-					ifaceFlowsMutex.Unlock()
-				})
+	for _, iface := range enable {
+		iface := iface
+		rg.Run(func() {
+
+			runCtx := logging.WithFields(ctx, slog.String("iface", iface))
+
+			cap := newCapture(iface, ifaces[iface]).SetSourceInitFn(cm.sourceInitFn)
+			if err := cap.run(runCtx); err != nil {
+				logging.FromContext(runCtx).Errorf("failed to start capture: %s", err)
+				return
 			}
-		}
-	} else {
-		for iface, mc := range cmCopy {
-			iface, mc := iface, mc
-			rg.Run(func() {
-				f := mc.capture.Flows()
-				ifaceFlowsMutex.Lock()
-				ifaceFlows[iface] = f
-				ifaceFlowsMutex.Unlock()
-			})
-		}
+
+			cm.captures.Set(iface, cap)
+		})
 	}
 	rg.Wait()
 
-	// convert data
-	var ifaceFlowInfos = make(map[string]capturetypes.FlowInfos, len(ifaceFlows))
-	for iface, flowLog := range ifaceFlows {
-		var flowInfos []capturetypes.FlowInfo
-		for _, flow := range flowLog.Flows() {
-			flowInfos = append(flowInfos, capturetypes.FlowInfo{
-				Idle:                    flow.HasBeenIdle(),
-				DirectionConfidenceHigh: flow.DirectionConfidenceHigh(),
-				Flow:                    flow.ToExtendedRow(),
-			})
+	// Disable any interfaces present in the negative list
+	for _, iface := range disable {
+		mc, exists := cm.captures.Get(iface)
+		if !exists {
+			continue
 		}
+		rg.Run(func() {
 
-		// sort by packets, descending
-		sort.SliceStable(flowInfos, func(i, j int) bool {
-			return flowInfos[i].Flow.Counters.SumBytes() > flowInfos[j].Flow.Counters.SumBytes()
-		})
+			runCtx := logging.WithFields(ctx, slog.String("iface", mc.iface))
 
-		ifaceFlowInfos[iface] = flowInfos
+			if err := mc.close(); err != nil {
+				logging.FromContext(runCtx).Errorf("failed to close capture: %s", err)
+				return
+			}
+
+			cm.captures.Delete(mc.iface)
+		})
 	}
+	rg.Wait()
 
-	return ifaceFlowInfos
+	logger.With("elapsed", time.Since(t0).Round(time.Millisecond).String()).Debug("updated interface list")
+	logger.With(
+		"elapsed", time.Since(t0).Round(time.Millisecond).String(),
+		"ifaces_added", enable,
+		"ifaces_removed", disable,
+	).Debug("updated interface configuration")
 }
 
-// ErrorsAll returns the error maps of all managed Capture instances.
-func (cm *Manager) ErrorsAll() map[string]ErrorMap {
-	errmapMutex := sync.Mutex{}
-	errormap := make(map[string]ErrorMap)
+// Close stops / closes all (or a set of) interfaces
+func (cm *Manager) Close(ctx context.Context, ifaces ...string) {
```

Comment:
Solved the lock conundrum for `Update()` and `Close()` by using `Update()` to perform the close, not the other way around. That way, `Update()` (or rather `update()`) can govern the global lock, thus prohibiting concurrent calls to those methods.

Reply:
TY! :pray: